### PR TITLE
Add optional "comment" field for emergency_shutoff

### LINKED ISSUE #1862: please provide an optional "comment" field for emergency_shutoff
Something that came up recently is it might be useful if we could document the reason for emergency shutoffs, more or less in the same way that we do for rules (or for tree closures in treestatus), typically by pointing to a bug number.  Then https://aus-api.mozilla.org/api/v1/emergency_shutoff/Firefox/nightly could be used by interested parties to know what's going on when they're not getting updates.

cc @Archaeopteryx

### PR DESCRIPTION
**Pull request by allan-silva**
- [x] Add comment field to emergency shutoff model
- [x] Add comment field to  api
- [x] Add comment field to Balrog UI
- [x] Do manual tests on ui
- [x] Check possible regression for SpeedDial, click fires on disabled elements. Or file a new issue...

Changes on balrogui:
image
image
image
image

Resolves #1862

### CODE REVIEW

File: ui/src/components/DialogAction/index.jsx [hunk 1]
```diff
@@ -95,10 +95,7 @@ function DialogAction(props) {
         {body && <DialogContentText component="div">{body}</DialogContentText>}
       </DialogContent>
       <DialogActions>
-        <Button
-          disabled={actionExecuting}
-          onClick={onClose}
-          action={actions => actions && actions.focusVisible()}>
```
Comment (allan-silva) [inline ui/src/components/DialogAction/index.jsx:101]: This behavior (`fucusVisible()`) makes typing the comment impossible. Seems not a big problem remove it. Let me know if has any other approach.

**Comment by allan-silva** [comment]
@jcristau My last contribution was long time ago, let me know if I missed any step.

**Review by gbrownmozilla — COMMENTED** [review]
This looks good to me. Thank you very much!

Since @jcristau originally opened the issue, I'd like him to give final approval here; he is on vacation currently - hope you don't mind waiting a couple of weeks.

**Comment by allan-silva** [comment]
@gbrownmozilla no problem.
Thanks!

File: tests/admin/views/test_emergency_shutoff.py [hunk 2]
```diff
@@ -38,19 +38,23 @@ def test_get_by_id(self):
...
+        self.assertIn("comment", data)
+        self.assertEqual(data["comment"], "Fennec panic!()")
 
     def test_get_notfound(self):
         resp = self._get("/emergency_shutoff/foo/bar")
         self.assertStatusCode(resp, 404)
 
     def test_create(self):
-        data = {"product": "Thunderbird", "channel": "release"}
+        data = {"product": "Thunderbird", "channel": "release", "comment": "Thunderbird panic!()"}
```
Comment (jcristau) [inline tests/admin/views/test_emergency_shutoff.py:49]: We should keep a test for POST to /emergency_shutoff without a comment, to make sure that still works.

**Review by jcristau — COMMENTED** [review]
A couple of nits but this looks great.  (Not approving just yet because I'd like to play with it locally, but I do want to get to it shortly.)

File: ui/src/components/DialogAction/index.jsx (hunk 1 above)
Comment (gabrielBusta) [reply ui/src/components/DialogAction/index.jsx:101]: I am not worried about it but there might be a reason it the code is calling the focus visible on that button. So I will take a closer look

Comment (gabrielBusta) [reply ui/src/components/DialogAction/index.jsx:101]: @jcristau this was focusing / highlighting the cancel button on the dialogs (i.e. the signing-off pop-ups) This was probably done so pressing enter would cancel the action / close the pop-up.

I think this change an improvement. Now we can tab through the buttons in the dialog and press enter to "click" on them, which is the way they should behave. Also the button wont take away the focus from other UI elements we might want to put in the dialogs (e.g. the text box on this PR)

Comment (gabrielBusta) [reply ui/src/components/DialogAction/index.jsx:101]: @jcristau this code was focusing / highlighting the cancel button on the dialogs (i.e. the signing-off pop-ups) This was probably done so pressing enter would cancel the action / close the pop-up.

I think this change is an improvement. Now we can tab through the buttons in the dialogs and press enter to "click" them (this is the way they should behave imo). Also the button wont take away the focus from other UI elements we might want to put in these dialogs (e.g. @allan-silva's text box)

Comment (gabrielBusta) [reply ui/src/components/DialogAction/index.jsx:101]: @jcristau this code was focusing / highlighting the cancel button on the dialogs (i.e. the signing-off pop-ups) This was probably done so pressing enter would cancel the action / close the pop-up.

I think this change is an improvement. Now we can tab through the buttons in the dialogs and press enter to "click" them (this is the way they should behave imo). Also the button wont take away the focus from other UI elements we might want to put in the body of these dialogs (e.g. @allan-silva's TextField)

**Review by gabrielBusta — CHANGES_REQUESTED** [review]
Looks good to me. Deferring to @jcristau for final approval here. Thank you!

**Review by gabrielBusta — APPROVED** [review]

File: ui/src/components/DialogAction/index.jsx (hunk 1 above)
Comment (allan-silva) [reply ui/src/components/DialogAction/index.jsx:101]: @gabrielBusta Thanks for take a close look on this!

**Review by jcristau — APPROVED** [review]
Looks great.  Thanks again @allan-silva!

**Pull request merged**